packages/driver/src/cy/commands/actions/selectFile.ts

```diff
@@ -37,8 +38,13 @@ const tryMockWebkit = (item) => {
 const createDataTransfer = (files: Cypress.FileReferenceObject[]): DataTransfer => {
   const dataTransfer = new DataTransfer()
 
-  files.forEach(({ contents, fileName = '', lastModified = Date.now() }) => {
-    const file = new File([contents], fileName, { lastModified })
+  files.forEach(({
+    contents,
+    fileName = '',
+    mimeType = mime.lookup(fileName) || '',
+    lastModified = Date.now(),
+  }) => {
+    const file = new File([contents], fileName, { lastModified, type: mimeType })
```

Comment:
We don't need any error handling/validation around the user-provided mimeTypes, right? Looks like the File constructor is happy with whatever you give it.

Reply:
That was my thinking as well.